### ISSUE TITLE
Handling of targeted data

### ISSUE THREAD

**Issue by Solafsson** [ggruenhagen3/coselenss#1]
Hi and thanks for this very useful package.

I've been testing it on some internal datasets using both WES and targeted data. I've had an issue with the targeted data which I'd like to report (very sorry if this is a user error). 
Whenever I use the sequenced.genes argument, the internal dndscv() function raises this error:
```
    if (!is.null(cv)) {  # Check compatibility of covariates with RefCDS
        if (!all(sapply(RefCDS, function(x) x$gene_name) == row.names(covs))) {
            stop("Row names in the covariates matrix do not coincide with the gene names in the reference database. Please, provide a compatible covariates matrix or run dNdScv with the option cv=NULL.") 
        }
    } 
```

It seems that the covs object is not subsetted to include just the genes in the sequenced.genes vector, causing a mismatch between the RefCDS and covs. The issue is fixed if one simply adds
`covs <- covs[rownames(covs) %in% gene_list,]`

before the second if statement. 

I figured since the paper mostly uses WES data from TCGA, maybe the package wasn't extensively tested using targeted data. Hope this is useful and not just the result of me misunderstanding the code. Thanks again for the package :)

**Comment by ggruenhagen3** [comment]
Hi @Solafsson, I'm glad the package has been helpful! You're right that the package wasn't extensively tested using targeted data. You're understanding seems correct and I will mark this as something to fix. Is the package still usable for you?

**Comment by Solafsson** [comment]
Yes, I've modified the function locally so it runs for now. Just wanted to make you aware of the issue.
